File: chatuikit/src/main/java/com/cometchat/chatuikit/shared/views/mediaviewer/CometChatImagePreviewUtils.java
package com.cometchat.chatuikit.shared.views.mediaviewer;

import android.view.ViewGroup;
import android.widget.ImageView;

public class CometChatImagePreviewUtils {

    public static CometChatImagePreview createImagePreview(ImageView imageView, ViewGroup container) {
        return CometChatImagePreview.create(imageView, container);
    }

    public static void setOnScaleChangedListener(CometChatImagePreview cometChatImagePreview, OnScaleChangedListener onScaleChangedListener) {
        cometChatImagePreview.setOnScaleChangedListener(onScaleChangedListener::onScaleChange);
    }

    public static void setOnViewTranslateListener(CometChatImagePreview cometChatImagePreview, OnViewTranslateListener onViewTranslateListener) {
        cometChatImagePreview.setOnViewTranslateListener(new CometChatImagePreview.OnViewTranslateListener() {
            @Override
            public void onStart(ImageView view) {
                onViewTranslateListener.onStart(view);
            }

            @Override
            public void onViewTranslate(ImageView view, float amount) {
                onViewTranslateListener.onViewTranslate(view, amount);
            }

            @Override
            public void onDismiss(ImageView view) {
                onViewTranslateListener.onDismiss(view);
            }

            @Override
            public void onRestore(ImageView view) {
                onViewTranslateListener.onRestore(view);
            }
        });
    }

    // Listener interface for scale change
    public interface OnScaleChangedListener {
        void onScaleChange(float scaleFactor, float focusX, float focusY);
    }

    // Listener interface for view translation events
    public interface OnViewTranslateListener {
        void onStart(ImageView view);

        void onViewTranslate(ImageView view, float amount);

        void onRestore(ImageView view);

        void onDismiss(ImageView view);
    }
}

File: chatuikit-kotlin/src/main/java/com/cometchat/uikit/kotlin/presentation/shared/mediaviewer/CometChatImagePreviewUtils.kt
package com.cometchat.uikit.kotlin.presentation.shared.mediaviewer

import android.view.ViewGroup
import android.widget.ImageView

/**
 * Utility object providing factory methods to create [CometChatImagePreview] instances
 * and convenience wrappers for setting listeners.
 */
object CometChatImagePreviewUtils {

    /**
     * Creates a [CometChatImagePreview] instance attached to the given [imageView] and [container].
     * The default drag-to-dismiss distance is set to 50% of the screen height via
     * [CometChatImagePreview.DEFAULT_DRAG_DISMISS_DISTANCE_IN_VIEW_HEIGHT_RATIO] in the
     * [CometChatImagePreview.onLayoutChange] callback.
     */
    @JvmStatic
    fun createImagePreview(imageView: ImageView, container: ViewGroup): CometChatImagePreview {
        return CometChatImagePreview.create(imageView, container)
    }

    /**
     * Sets an [OnScaleChangedListener] on the given [cometChatImagePreview].
     */
    @JvmStatic
    fun setOnScaleChangedListener(
        cometChatImagePreview: CometChatImagePreview,
        onScaleChangedListener: OnScaleChangedListener
    ) {
        cometChatImagePreview.setOnScaleChangedListener(object : CometChatImagePreview.OnScaleChangedListener {
            override fun onScaleChange(scaleFactor: Float, focusX: Float, focusY: Float) {
                onScaleChangedListener.onScaleChange(scaleFactor, focusX, focusY)
            }
        })
    }

    /**
     * Sets an [OnViewTranslateListener] on the given [cometChatImagePreview].
     */
    @JvmStatic
    fun setOnViewTranslateListener(
        cometChatImagePreview: CometChatImagePreview,
        onViewTranslateListener: OnViewTranslateListener
    ) {
        cometChatImagePreview.setOnViewTranslateListener(object : CometChatImagePreview.OnViewTranslateListener {
            override fun onStart(view: ImageView) {
                onViewTranslateListener.onStart(view)
            }

            override fun onViewTranslate(view: ImageView, amount: Float) {
                onViewTranslateListener.onViewTranslate(view, amount)
            }

            override fun onDismiss(view: ImageView) {
                onViewTranslateListener.onDismiss(view)
            }

            override fun onRestore(view: ImageView) {
                onViewTranslateListener.onRestore(view)
            }
        })
    }

    /**
     * Listener interface for scale change events.
     */
    interface OnScaleChangedListener {
        fun onScaleChange(scaleFactor: Float, focusX: Float, focusY: Float)
    }

    /**
     * Listener interface for view translation events during drag-to-dismiss gestures.
     */
    interface OnViewTranslateListener {
        fun onStart(view: ImageView)
        fun onViewTranslate(view: ImageView, amount: Float)
        fun onRestore(view: ImageView)
        fun onDismiss(view: ImageView)
    }
}
